src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -253,25 +287,29 @@ public void shouldNotCreateOrganizationIfConfidential() {
         Organization organization = generateNewOrganization(generateExistingAccount());
         organization.setIsConfidential(true);
 
-        int dbSize = organizationService.findAllDTOs().size();
-        importService.createOrUpdateOrganization(organization, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
+        var created = importManager.createOrUpdateOrganization(organization, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(dbSize, organizationService.findAllDTOs().size());
+        assertEquals(0, organizationService.findAllDTOs().size());
+        assertNull(created);
     }
 
     @Test
     @Transactional
     public void shouldNotUpdateOrganizationIfConfidential() {
         SystemAccount account = generateExistingAccount();
-        Organization newOrganization = generateNewOrganization(account);
-        newOrganization.setIsConfidential(true);
-        generateExistingOrganization(account);
+        em.persist(account);
+        Organization newOrganization = generateExistingOrganization(account);
+        importManager.createOrUpdateOrganization(newOrganization, EXISTING_EXTERNAL_ID, PROVIDER,
+            new DataImportReport());
+        assertEquals(1, organizationService.findAllDTOs().size());
 
-        int dbSize = organizationService.findAllDTOs().size();
-        Organization updated = importService.createOrUpdateOrganization(newOrganization, EXISTING_EXTERNAL_ID, PROVIDER,
+        newOrganization.setIsConfidential(true);
+        newOrganization.setName(NEW_STRING);
+        var updated = importManager.createOrUpdateOrganization(newOrganization, EXISTING_EXTERNAL_ID, PROVIDER,
             new DataImportReport());
 
-        assertEquals(dbSize, organizationService.findAllDTOs().size());
+        assertEquals(1, organizationService.findAllDTOs().size());
+        assertEquals(EXISTING_STRING, organizationService.findAllDTOs().get(0).getName());
```

Comment:
In my opinion, this is misleading because the organization is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.

Reply:
@oskarhinc can you create that task?

Reply:
Sure. There is something similar in the backlog so I've added a new comment there: https://github.com/benetech/ServiceNet/issues/220